### PR TITLE
dotCMS/core#16765 Create dot-form-error-message

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx [hunk 1]
```diff
@@ -113,9 +132,33 @@ export class DotFormComponent {
...
-            ...this.value
-        });
+
+        fetch(SUBMIT_FORM_API_URL, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                stName: this.variable || 'ContactUs',
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: why `ContactUS`

File: projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx [hunk 2]
```diff
@@ -113,9 +132,33 @@ export class DotFormComponent {
...
+                        status: response.status
+                    };
+                    throw error;
+                }
+                return response.text();
+            })
+            .then((_text: string) => {
+                // Go to success page
+            })
+            .catch(({ message, status }: DotCMSFormSubmitError) => {
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: type?

File: projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx (hunk 1 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: No... this was just for testing.. needs to remove.

File: projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx (hunk 2 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: `DotCMSFormSubmitError`

File: projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx [hunk 3]
```diff
@@ -113,9 +132,33 @@ export class DotFormComponent {
...
+                    const error: DotCMSFormSubmitError = {
+                        message: await response.text(),
+                        status: response.status
+                    };
+                    throw error;
+                }
+                return response.text();
+            })
+            .then((_text: string) => {
+                // Go to success page
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: where or who sets the `success` page?

Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx]: Screen Shot 2019-07-22 at 11 31 49 AM

**Pull request merged**